addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/Spring4ShellScanRuleUnitTest.java

```diff
@@ -0,0 +1,268 @@
+                    }
+                });
+        HttpMessage msg = getHttpMessage("GET", path, "");
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+        assertThat(httpMessagesSent, hasSize(2));
+    }
+
+    @Test
+    void shouldAlertOnVulnerableGet() throws HttpMalformedHeaderException {
+        // Given
+        String path = "/getvulnerable";
+        this.nano.addHandler(
+                new NanoServerHandler(path) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        if (NanoHTTPD.Method.GET.equals(session.getMethod())
+                                && session.getQueryParameterString()
+                                        .contains(Spring4ShellScanRule.ATTACK)) {
+                            this.consumeBody(session);
+                            return newFixedLengthResponse(
+                                    Response.Status.BAD_REQUEST, NanoHTTPD.MIME_HTML, "");
+                        }
+                        this.consumeBody(session);
+                        return newFixedLengthResponse(Response.Status.OK, NanoHTTPD.MIME_HTML, "");
+                    }
+                });
+        HttpMessage msg = getHttpMessage("GET", path, "");
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertThat(alertsRaised.get(0).getName(), is(equalTo("Spring4Shell")));
+        assertThat(alertsRaised.get(0).getRisk(), is(equalTo(3)));
+        assertThat(alertsRaised.get(0).getConfidence(), is(equalTo(2)));
+        assertThat(alertsRaised.get(0).getEvidence(), is(equalTo("HTTP/1.1 400 Bad Request ")));
+        assertThat(alertsRaised.get(0).getAttack(), is(equalTo(Spring4ShellScanRule.ATTACK)));
+    }
+
+    @Test
+    void shouldAlertOnVulnerablePost() throws HttpMalformedHeaderException {
+        // Given
+        String path = "/postvulnerable";
+        this.nano.addHandler(
+                new NanoServerHandler(path) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        if (NanoHTTPD.Method.POST.equals(session.getMethod())
+                                && this.getBody(session).contains(Spring4ShellScanRule.ATTACK)) {
+                            this.consumeBody(session);
+                            return newFixedLengthResponse(
+                                    Response.Status.BAD_REQUEST, NanoHTTPD.MIME_HTML, "");
+                        }
+                        this.consumeBody(session);
+                        return newFixedLengthResponse(Response.Status.OK, NanoHTTPD.MIME_HTML, "");
+                    }
+                });
+        HttpMessage msg = getHttpMessage("GET", path, "");
```

Comment:
POST?

Reply:
No, deliberately testing that we try a POST evemn though the original is a get :)